modules/nf-core/deepvariant/makeexamples/main.nf

```diff
@@ -0,0 +1,64 @@
+process DEEPVARIANT_MAKEEXAMPLES {
+    tag "$meta.id"
+    label 'process_high'
+
+    //Conda is not supported at the moment
+    container "nf-core/deepvariant:1.6.1"
+
+    input:
+    tuple val(meta), path(input), path(index), path(intervals)
+    tuple val(meta2), path(fasta)
+    tuple val(meta3), path(fai)
+    tuple val(meta4), path(gzi)
+
+    output:
+    tuple val(meta), path("${prefix}.examples.tfrecord-*-of-*.gz{,.example_info.json}"),    emit: examples
+    tuple val(meta), path("${prefix}.gvcf.tfrecord-*-of-*.gz"),        emit: gvcf
+    path "versions.yml",  emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    // Exit if running this module with -profile conda / -profile mamba
+    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
+        error "DEEPVARIANT module does not support Conda. Please use Docker / Singularity / Podman instead."
+    }
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def regions = intervals ? "--regions ${intervals}" : ""
+
+    """
+    seq 0 ${task.cpus - 1} | parallel -q --halt 2 --line-buffer /opt/deepvariant/bin/make_examples \\
+        --mode calling \\
+        --ref "${fasta}" \\
+        --reads "${input}" \\
+        --examples "./${prefix}.examples.tfrecord@${task.cpus}.gz" \\
+        --channels "insert_size" \\
```

Comment:
https://github.com/nf-core/modules/pull/5990#discussion_r1687551686

TODO

Reply:
> [fellen31](https://github.com/fellen31) [3 weeks ago](https://github.com/nf-core/modules/pull/6172#discussion_r1687551686)
Suggested change
        --channels "insert_size" \\
This needs to go into config, since it's not active for HYBRID_PACBIO_ILLUMINA, PACBIO or ONT_R104 model types.
Member
Author
@[fa2k](https://github.com/fa2k) fa2k [yesterday](https://github.com/nf-core/modules/pull/6172#discussion_r1715335472)
Yes according to the guidelines, the arguments to the tool should be minimal and anything extra put in the config. So I agree with this, but I fear that leaving out insert_size could make it very easy to misconfigure the tool for users. What could be done to help users remember to apply this channel using config when appropriate?

> Do you reckon this should go into ext.args or should I create a dedicated config variable for insert size?

I see your point, the wgs/wes models will be used most, but by hardcoding `--channels "insert_size"` in the module  you will not be able to run the long-read/hybrid models at all. 

Who are the users here? As I see it, a pipeline will have this subworkflow with a config. The subworkflow takes a val(model_type), and the settings for make_examples is determined by this, either by logic in the subworkflow config, or model_type is passed on to makeexamples and the logic is in the module. The burden is either pushed a bit more towards the pipeline maintainers, or the module maintainers. 

> What could be done to help users remember to apply this channel using config when appropriate?

I can see the point of having the logic in the module, even though it might stray from the guidelines. 

The end user of a pipeline should still be able to use a simple parameter or the samplesheet  to choose between wes or wgs analysis.  

> Do you reckon this should go into ext.args or should I create a dedicated config variable for insert size?

If you go with having the logic in config, I think the consensus is to use `ext.args`, and not have a dedicated config variables.

Reply:
By keeping the current deepvariant as rundeepvariant as well, pipeline maintainers still have the option to go with the simple solution where you don't need to care about knowing the correct makeexamples parameters for a certain model.

Reply:
I've removed --channels from makeexamples, and adding a note to the README in modules/nf-core/deepvariant. For other data types it may be more complex to see what channels are needed.

I kind of agree with your reasoning that it's the pipeline maintainer's responsibility. (As a pipeline user I personally find the pipeline's config dir difficult to read, when trying to understand a pipeline, but I'm slowly learning to always look in the config.)